researchspace/researchspace
# oxigraph. Ability to enable silent mode for repository.

# Why

Some RDF databases, like https://github.com/oxigraph/oxigraph return error if one tries to clear graph that doesn't exist. That is acceptable behavior according to standard. 

# What

This change introduce new repository config property `silentMode`. When the value is true, RDF4J will generate `CLEAR SILENT GRAPH` instead of `CLEAR GRAPH` that is generated by default. `silentMode` is false by default and need to be explicitly enabled in the repository config. So it is non breaking change and should not have any side effect.

```
@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
@prefix sparql: <http://www.openrdf.org/config/repository/sparql#> .
@prefix rep: <http://www.openrdf.org/config/repository#> .
@prefix sail: <http://www.openrdf.org/config/sail#> .
@prefix sr: <http://www.openrdf.org/config/repository/sail#> .
@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
@prefix mph: <http://www.researchspace.org/resource/system/repository#> .
@prefix ephedra: <http://www.researchspace.org/resource/system/ephedra#> .
@prefix fedsail: <http://www.openrdf.org/config/sail/federation#> .

[] a rep:Repository;
    rep:repositoryID "default";
    rdfs:label "Default G17 HTTP SPARQL Repository";
    rep:repositoryImpl [
        rep:repositoryType "researchspace:SPARQLRepository";
    sparql:query-endpoint <http://oxigraph:7878/query>;
    sparql:update-endpoint <http://oxigraph:7878/update>;
    mph:quadMode true;
    mph:writable true ;
    mph:silentMode true ;
    
    ] .

```

# Meta
Along the way I found a bug in RepositoryManager that prevented initialization when default repository was initialized with `default.ttl` and there is a custom repository that proxies requests to default. It is also fixed.

# How To Test

see https://github.com/researchspace/researchspace-docker-compose/issues/3 here setup with oxigraph.
